### ISSUE TITLE
[Connect] User is not redirected after leaving a room, leading to ambiguity and blank error

### ISSUE THREAD

**Issue by wafflewitch** [forem/forem#9502]
**Describe the bug**

In both full screen (/connect route) and sidecar view, leaving a room does not result in redirecting the user away from that room. This leaves the user in an ambiguous situation where they can still see the room they have left. However, if the user tries to send a message in the left room, a blank error appears.

**To Reproduce**

**Expected behavior**

Upon leaving a room, the user should be redirected (or the page should be refreshed) to officially remove the user from the room and prevent the ambiguous situation described above. 

Also, the error should be improved to display a message. 

Finally, I'm also wondering if the error is supposed to fit the user's chosen color theme... ?)

**Screenshots**

Full screen view:
![dev_connect_full-scr_left-room-empty-err](https://user-images.githubusercontent.com/27379918/88437048-405b4580-cdcb-11ea-8960-e985bf8ebab6.gif)

Sidecar view:
![dev_connect_sidecar_left-room-empty-err](https://user-images.githubusercontent.com/27379918/88437329-de4f1000-cdcb-11ea-9db3-3696635dbb38.gif)

**Desktop (please complete the following information):**

- OS: Mac
- Browser: Chrome
- Version: 83.0.4103.116

**Comment by juliannatetreault** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @forem/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/forem/forem/blob/master/CONTRIBUTING.md).

**Comment by AlbertoPdRF** [comment]
I'm looking into this one! I may come back with a few questions later on 😛

**Comment by nickytonline** [comment]
Sounds good @AlbertoPdRF. I've assigned it to you.

**Comment by AlbertoPdRF** [comment]
First of all, sorry for the late follow up! So, if I got everything correctly, when a user clicks on the button to leave a channel, and the request succeeds, [this is the code that gets executed](https://github.com/forem/forem/blob/master/app/javascript/chat/ChatChannelSettings/ChatChannelSettings.jsx#L311):

https://github.com/forem/forem/blob/538c083d123f1437e35bce14d9aa08d16fbfc177/app/javascript/chat/ChatChannelSettings/ChatChannelSettings.jsx#L57-L89

But instead of that, which won't work, a redirect should happen. I would like to get some direction about how to handle the redirect: should it happen on the RoR or on the Preact side? Depending on the answer, I may have more questions, so if you can give a bit of context that would be nice 🙂

**Comment by sarthology** [comment]
Hey @AlbertoPdRF,
Sorry for late reply, Yes you got it right.
So a quick fix can be just reload (which is equivalent to redirecting via ROR).
A better way will be handle it in Preact, By fetching all the channels again  using `getChannels` function and reseting the active channel. You may have to do that in `chat.jsx`, see this code snippet for help
```Javascript 
  handleUpdateRequestCount = (isAccepted = false, acceptedInfo) => {
    if (isAccepted) {
      const searchParams = {
        query: '',
        retrievalID: null,
        searchType: '',
        paginationNumber: 0,
      };
      getChannels(searchParams, 'all', this.loadChannels);
      this.triggerSwitchChannel(
        parseInt(acceptedInfo.channelId, 10),
        acceptedInfo.channelSlug,
      );
    }
```

Let me know if need further help. 😊
May the Source be with you. 🖖🏼🔥

**Comment by AlbertoPdRF** [comment]
It turns out #9804 also fixed this issue, I just found out when trying to rebase [my branch with the fix](https://github.com/MLH-Fellowship/dev.to/tree/AlbertoPdRF/redirect-after-leaving-chat-9502) on top of `forem/master` before submitting a PR.

**Comment by narender2031** [comment]
This issue wad closed in #9804
